pyvrp/tests/test_Population.py

```diff
@@ -177,24 +178,43 @@ def test_select_returns_same_parents_if_no_other_option():
 # // TODO test more select() - diversity, feas/infeas pairs
 
 
-def test_restart_generates_min_pop_size_new_individuals():
-    """
-    Tests if restarting the population will generate ``min_pop_size`` new
-    individuals.
-    """
-    data = read("data/RC208.txt", "solomon", "dimacs")
+def test_same_initial_solutions():
+    data = read("data/E-n22-k4.txt", round_func="round")
     pm = PenaltyManager(data.vehicle_capacity)
     rng = XorShift128(seed=12)
 
-    params = PopulationParams()
-    pop = Population(data, pm, rng, broken_pairs_distance, params)
+    params = PopulationParams(min_pop_size=10)
+    init = make_random_solutions(params.min_pop_size, pm, rng, data)
+    pop = Population(init, bpd, params)
+
+    # Check that the initial population individuals have the same routes as the
+    # passed-in initial solutions. We check for equality here because the
+    # population creates new individuals.
+    for indiv in pop:
+        assert_(np.any(indiv == other for other in init))
 
-    old = {individual for individual in pop}
     pop.restart()
-    new = {individual for individual in pop}
 
-    assert_equal(len(pop), params.min_pop_size)
-    assert_equal(len(old & new), 0)  # no old pops survived the restart
+    # Check again for the restarted population.
+    for indiv in pop:
+        assert_(np.any(indiv == other for other in init))
+
+
+@mark.parametrize("num_init", [1, 15, 21, 30, 100])
+def test_num_initial_solutions(num_init):
+    data = read("data/OkSmallFeasible.txt")
+    pm = PenaltyManager(data.vehicle_capacity)
+    rng = XorShift128(seed=12)
+
+    params = PopulationParams(min_pop_size=0, generation_size=10)
+    init = make_random_solutions(num_init, pm, rng, data)
+    pop = Population(init, bpd, params)
+
+    # If there are more initial individuals than the maximal population size
+    # allows, then a purge is triggered. This resulting population size is
+    # equal to ``num_init`` modulo the number individuals that need to
+    # be added to trigger a purge (11).
+    assert_equal(len(pop), num_init % 11)
```

Comment:
This isn't true in general. Suppse we add `15` individuals. Then `15 % 11 == 4`. But what if we've added 7 feasible and 8 infeasible solutions? Then no purge would have been triggered because neither subpopulation would have been of maximal size.

The reason this test passes is that all solutions are feasible in this instance, which is not explicitly mentioned in the test. Can we document that much more explicitly?